### PR TITLE
[Cherry-pick][JIT] Add Predictor for JITLayer (#47379)

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

New features
### PR changes

Others
### Describe

[Cherry-pick][JIT] Add Predictor for JITLayer (#47379)

### CODE REVIEW

**Review by yeliang2258 — APPROVED** [review]
LGTM

File: paddle/fluid/inference/api/analysis_predictor.cc [hunk 1]
```diff
@@ -168,20 +168,27 @@ bool PaddleTensorToLoDTensor(const PaddleTensor &pt,
...
-      paddle::platform::errors::Fatal(
-          "Cannot convert to LoDTensor because LoDTensor creation failed."));
-  PADDLE_ENFORCE_NOT_NULL(
-      pt.data.data(),
-      paddle::platform::errors::InvalidArgument(
-          "The data contained in the input PaddleTensor is illegal."));
+  // NOTE(Aurelius84): Some kernels support zero shape input
+  // without memory holder, we should skip enforce logic.
+  bool has_zero_dim = (phi::product(ddim) == 0);
+  if (has_zero_dim) {
```
Comment (zh794390558) [inline paddle/fluid/inference/api/analysis_predictor.cc:174]: !has_zero_dim ?

File: paddle/fluid/jit/engine/predictor_engine.cc [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+std::vector<Tensor> PredictorEngine::operator()(
+    const std::vector<Tensor> &inputs) {
+  auto dense_tensors = utils::ToDenseTensors(inputs);
+  return utils::ToTensors(this->operator()(dense_tensors));
+}
+
+std::vector<DenseTensor> PredictorEngine::operator()(
+    const std::vector<DenseTensor> &inputs) {
+  for (auto t : inputs) {
+    VLOG(1) << "inputs is init: " << t.initialized();
```
Comment (zh794390558) [inline paddle/fluid/jit/engine/predictor_engine.cc:63]: VLOG的级别提升下吧

File: paddle/fluid/inference/api/analysis_predictor.cc (hunk 1 above)
Comment (Aurelius84) [reply paddle/fluid/inference/api/analysis_predictor.cc:174]: 在下个PR里

File: paddle/fluid/jit/engine/predictor_engine.cc (hunk 2 above)
Comment (Aurelius84) [reply paddle/fluid/jit/engine/predictor_engine.cc:63]: 见：https://github.com/PaddlePaddle/Paddle/pull/47428/files

**Review by zh794390558 — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM for const_cast

**Review by zhiqiu — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**